cloud/services/object_storage_objects.go

```diff
@@ -0,0 +1,135 @@
+package services
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	s3manager "github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/smithy-go"
+	"github.com/go-logr/logr"
+
+	"github.com/linode/cluster-api-provider-linode/cloud/scope"
+)
+
+func CreateObject(ctx context.Context, mscope *scope.MachineScope, data []byte, logger logr.Logger) (string, error) {
+	logger.Info("Create Object Storage object")
+
+	if mscope == nil {
+		return "", errors.New("machine scope can't be nil")
+	}
+
+	if mscope.S3Client == nil {
+		return "", errors.New("nil S3 client in machine scope")
+	}
+
+	if mscope.LinodeCluster.Spec.ObjectStore == nil {
+		return "", errors.New("nil cluster object store")
+	}
+
+	if len(data) == 0 {
+		return "", errors.New("got empty data")
+	}
+
+	bucket, err := mscope.GetBucketName(ctx)
+	if err != nil || bucket == "" {
+		return "", errors.New("no bucket name")
```

Comment:
Could we define these in a var block up top similar to internal/controller/linodemachine_controller_helpers.go? Might be useful for tests to have these defined.

Reply:
Might also make sense to create a helper function to check all the required variables are set and that we can get the bucket since there seems to be some overlap in the DeleteObject function

Reply:
> define these errors in a var block
QQ: Why do we want(?) this level of insight into the implementation details in our unit tests?

> create a helper function to check all the required variables are set
Sounds good!

Reply:
I didn't want to litter the package with error globals here, so I just matched the error string in the tests 😇